src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java

```diff
@@ -345,7 +345,7 @@ public long getTotalDirCount() throws MopsException {
     @Override
     @SuppressWarnings({ "PMD.AvoidReassigningParameters", "PMD.LawOfDemeter" })
     public Directory renameDirectory(Account account, long dirId, String newName) throws MopsException {
-        if (newName.isEmpty()) {
+        if (newName.replaceAll("\\s+", "").isEmpty()) {
```

Comment:
can maybe use String.trim()

Reply:
String.trim() only removes trailing white spaces. .replaceAll("\\s+", "") removes all non printable characters. .trim() would leave a \t as filename, which I would consider empty.

Reply:
fair

Reply:
I just realized that a tab char or white space would simply be replaced by an underscore. Maybe we could completely ignore non printable characters as they'll be replaced anyway :thinking:

Reply:
\s will match any whitespace character ´[ \t\n\x0B\f\r]´, while `.trim()` will remove every leading or trailing ascii char with codepoint <= 0x20. That is every control character and the space character. This includes all non printable characters except for `DEL` at the end of the ascii table.

but newName.isEmpty() should be enough, as every odd character gets replaced either way 🤷‍♀

Reply:
It's gone now completely.